medical/models/medical_patient.py

```diff
@@ -95,6 +95,15 @@ def action_invalidate(self):
             ('z', 'law marriage'),
         ])
 
+    @api.multi
+    def onchange_state(self, state_id):
+        '''
+        Hack to allow for onchange_state on partner.
+        Uses current user's partner, as there should be no correlation
+        between the partner and the response from this method anyways
+        '''
+        return self.env.user.partner_id.onchange_state(state_id)
```

Comment:
I tried and it seemed like it wasn't compatible with the `res.partner` onchange because of the param - on second look, I just needed to refactor to use the pseudo-record. Good call!